pkg/executables/govc.go

```diff
@@ -315,14 +331,13 @@ func (g *Govc) DeployTemplateFromLibrary(ctx context.Context, templateDir, templ
 		// checks based on the label.
 		disk1 := ""
 		disk2 := ""
-		for _, deviceInfo := range devicesInfo.([]interface{}) {
-			deviceMetadata := deviceInfo.(map[string]interface{})["DeviceInfo"]
-			deviceLabel := deviceMetadata.(map[string]interface{})["Label"].(string)
+		for _, device := range devicesInfo {
+			deviceLabel := device.DeviceInfo.Label
 			// Get the name of the hard disk and resize the disk to 20G
 			if strings.EqualFold(deviceLabel, "Hard disk 1") {
-				disk1 = deviceInfo.(map[string]interface{})["Name"].(string)
+				disk1 = device.Name
 			} else if strings.EqualFold(deviceLabel, "Hard disk 2") {
-				disk2 = deviceInfo.(map[string]interface{})["Name"].(string)
+				disk2 = device.Name
 				break
```

Comment:
why is the break within the 'if else'? If the idea is that after the disk#2 is found we want to break, then are we sure it  is returned in sequence?

Reply:
Also the log message below on line 345/360 for the error case during resizeDisk needs to be fixed, it currently says 20G.. But disk2 is being resized to 22G. Need to use diskSizeInGB var there.

Reply:
This is the same logic we had before. I think the idea is that if Disk 2 is there, then we want to resize disk 2. So it doesn't matter if there are other disks as well.

Reply:
> Also the log message below on line 345/360 for the error case during resizeDisk needs to be fixed, it currently says 20G.. But disk2 is being resized to 22G. Need to use diskSizeInGB var there.

fixed